### ISSUE TITLE
Parts duplicated and others no longer available.

### ISSUE THREAD

**Issue by alexanderreichard** [tomasero/openSleep#6]
The 1X2-SMD-HEADER, 1X3-SMD-HEADER, and 1X5-SMD-HEADER, all link to the same Digikey page. Is this intentional? 
There doesn't seem to be links on that page to differentiate the headers. 

Additionally the RFD22301, is no longer available from the link provided.

**Comment by tomasero** [comment]
You’re right, they’re the same component. I made the distinction cause you’ll have to cut the headers in different sizes to fit the appropriate holes.

Regarding the RFD22301, you’re also right. That was a nice NRF52-based module because its not that big and super easy to solder, but unfortunately it's no longer available. We built a second revision that uses a different NRF52-based module (BC832). We will post an update soon.

Please let me know if you have any other question.

**Comment by froslie** [comment]
Would it be possible to provide link for the 32.768 crystal use in the second revision? Looking forward to the update on this version.

**Comment by evwimble** [comment]
@froslie I think this may work: https://www.digikey.com/short/zbw71m

I don't mean to hijack this thread but wondering if we could get a complete parts list for v2, like is available for v1 :) I'm happy to share what I've found so far when I get back to my computer. I'm mostly stumped on the specifics of the capacitors & resistors, and how the FTDI, JLINK & Tag-Connect all fit together for programming.

**Comment by FEckeroth** [comment]
Is there an update to the hardware needed? Would be very nice if there was links to it as the version 1

**Comment by jsharf** [comment]
Hey! I recently built a V2 HW, and found finding all the parts on digikey to take several hours. Here are the parts I used:

https://docs.google.com/spreadsheets/d/1bM1ne3_NdHfvHR-6b3vplZjLTPfxBPxb_ulYGgKfk5I/edit?usp=sharing

I don't have time at the moment to add this to the github, but maybe someone else can? Also, some of the parts are out of stock (especially the battery connector). I found that by searching for similar parts I was able to find suitable replacements that are available immediately (for example for the battery connector, it's a connector made by JST and it's their "PH" connector line with two pins. It looks like the only kind left that's in stock is the through-hole connector. I just bent the pins into place and soldered it down, and then added some glue for structural reinforcement:

https://www.digikey.com/product-detail/en/jst-sales-america-inc/B2B-PH-K-S-LF-SN/455-1704-ND/926611

**Comment by jsharf** [comment]
Oh also you'll need some gear to program it. If you want the official route, you can get the NRF52832 dev kit and a matching tag connector cable (I didn't do the research to find the *exact* right one but it'll need to be a TC2030 series)

The dev kit is expensive so I'm just using a cheap $4 stlink V2 dongle I found on amazon. I'm just using it as-is for now, but you might also be interested in knowing that apparently it's possible to load the black magic probe firmware on them and get a much better debugging/flashing experience.

And then you'll need a tagconnector programming cable. I used this one, and I have some jumper wires to connect it to the Stlink V2 dongle:

https://www.tag-connect.com/product/tc2030-idc-nl

**Comment by FEckeroth** [comment]
Thank you so much for the info!

How is the device
Working for you?

mån 28 sep. 2020 kl. 18:04 skrev Jacob Sharf <notifications@github.com>:

**Comment by jsharf** [comment]
Admittedly I haven't actually programmed it yet. I built the device two weeks ago, but I didn't have the programming cable yet. It came in this week and I haven't had a chance to flash it yet since I've been busy at work. But the built-in factory firmware at least shows up on a bluetooth scan from my phone, so I know I got most of the board right :)

I'll probably give flashing it a shot tonight, and then I can sleep with the dormio. Will give an update tomorrow.

**Comment by FEckeroth** [comment]
I will look forward for your update!

mån 28 sep. 2020 kl. 20:28 skrev Jacob Sharf <notifications@github.com>:

**Comment by froslie** [comment]
Thank you for communicating on this - I also look forward to your update on programming it. I finished building one this past weekend. Assuming the parts I used are correct, I will program when you've had chance to share you experience. Much appreciated!

**Comment by jsharf** [comment]
Hey -- I just wanted to share my experience programming the device since it's been over two weeks since my last update. In short... I had some issues. Hopefully I can resolve them soon..

I built & flashed the bootloader from here: https://github.com/adafruit/Adafruit_nRF52_Bootloader
I then used my stlink + openocd to flash the bootloader

Unfortunately, at this point I got stuck. No matter how hard I tried, the adafruit-nrfutil tool wouldn't program the bootloader.

After some debugging, I discovered it was because I hadn't successfully soldered the UART Rx/Tx. They're on the BGA portion of the NRF52832 and the contacts didn't quite connect (i confirmed this by configuring them to GPIO outputs and then seeing if I saw any voltage on the RX/TX pins -- I didn't).

I then reflowed the chip and was able to program the application onto the board using adafruit-nrfutil. Unfortunately I ended up going down a rabbit hole of minor tweaks to the board. I was cleaning up some solder in the via underneath the RX pin and this caused solder on the BGA pad to escape down the via, so now I can't program it again.

During the brief period where it was working, the application would hardfault in bleSetup() in the Arduino setup() routine. I think I just need to make sure I'm working with a compatible softdevice (nordic's bluetooth stack). This is flashed via the adafruit-nrfutil utility. I'm going to try reflowing again tomorrow, I hope that works...

**Comment by FEckeroth** [comment]
Did you get it to work?

**Comment by jsharf** [comment]
Unfortunately I did not. I continued having issues with the pins underneath the microcontroller.

**Comment by akshar001** [comment]
@jsharf do we need to use amplifier along with ecg probes or not
